### PR TITLE
Use machine/v1alpha1 from openshift/api

### PR DESCRIPTION
**Pull request by pierreprinetti**
Implements [OSASINFRA-2981](https://issues.redhat.com//browse/OSASINFRA-2981)

Depends on https://github.com/openshift/api/pull/1129

This PR is based on #29, but does not implement the changes to ProjectID/TenantID which we'll tackle separately

### CODE REVIEW

File: pkg/machine/convert.go [hunk 1]
```diff
@@ -62,7 +63,30 @@ func clusterProviderStatusToClusterStatus(cps *openstackconfigv1.OpenstackCluste
 
 // Looks up a subnet in openstack and gets the ID of the network its attached to
 func getNetworkID(filter *openstackconfigv1.SubnetFilter, networkService *networking.Service) (string, error) {
-	listOpts := subnets.ListOpts(*filter)
+	// listOpts := subnets.ListOpts(*filter)
```
Comment (mandre) [inline pkg/machine/convert.go:66]: Probably something you forgot to remove.

Comment (pierreprinetti) [reply pkg/machine/convert.go:66]: :( indeed

**Comment by pierreprinetti** [comment]
e2e-openstack failed once because it failed to deploy and twice because of failed tests. I am not sure there's some amount of consistency in the test failures.

/retest

**Comment by MaysaMacedo** [comment]
@pierreprinetti  Should these files be updated [apis.go](https://github.com/openshift/machine-api-provider-openstack/blob/master/pkg/apis/apis.go#L24) and [register.go](https://github.com/openshift/machine-api-provider-openstack/blob/master/pkg/apis/openstackproviderconfig/v1alpha1/register.go#L29)?

**Comment by pierreprinetti** [comment]
> @pierreprinetti Should these files be updated [apis.go](https://github.com/openshift/machine-api-provider-openstack/blob/master/pkg/apis/apis.go#L24) and [register.go](https://github.com/openshift/machine-api-provider-openstack/blob/master/pkg/apis/openstackproviderconfig/v1alpha1/register.go#L29)?

The platform provider spec for OpenStack is still defined here in MAPO, so I believe that there should still exist register logic for this group.

Maybe @mdbooth can help us here

**Comment by mdbooth** [comment]
> > @pierreprinetti Should these files be updated [apis.go](https://github.com/openshift/machine-api-provider-openstack/blob/master/pkg/apis/apis.go#L24) and [register.go](https://github.com/openshift/machine-api-provider-openstack/blob/master/pkg/apis/openstackproviderconfig/v1alpha1/register.go#L29)?
> 
> The platform provider spec for OpenStack is still defined here in MAPO, so I believe that there should still exist register logic for this group.
> 
> Maybe @mdbooth can help us here

I vaguely recall that we didn't move it to openshift/api because nothing is using it. Try deleting it and see if it breaks anything important. If not, get it gone!

**Comment by pierreprinetti** [comment]
> > > @pierreprinetti Should these files be updated [apis.go](https://github.com/openshift/machine-api-provider-openstack/blob/master/pkg/apis/apis.go#L24) and [register.go](https://github.com/openshift/machine-api-provider-openstack/blob/master/pkg/apis/openstackproviderconfig/v1alpha1/register.go#L29)?
> > 
> > 
> > The platform provider spec for OpenStack is still defined here in MAPO, so I believe that there should still exist register logic for this group.
> > Maybe @mdbooth can help us here
> 
> I vaguely recall that we didn't move it to openshift/api because nothing is using it. Try deleting it and see if it breaks anything important. If not, get it gone!

ok, I removed the whole thing in a separate commit, ready to be reverted if something blows up.

**Comment by pierreprinetti** [comment]
/test e2e-openstack
No need to retest NFV, as it is currently broken

File: pkg/machine/convert.go [hunk 2]
```diff
@@ -316,7 +353,7 @@ func MachineToInstanceSpec(machine *machinev1.Machine, apiVIP, ingressVIP, userD
 			ProjectID:           port.ProjectID,
 			SecurityGroups:      port.SecurityGroups,
 			AllowedAddressPairs: make([]capov1.AddressPair, len(port.AllowedAddressPairs)),
-			HostID:              port.HostID,
+			HostID:              port.DeprecatedHostID,
```
Comment (mdbooth) [inline pkg/machine/convert.go:356]: We should ditch this. It's not a usable feature.

**Comment by pierreprinetti** [comment]
/test e2e-openstack
A second try before digging in

**Comment by mandre** [comment]
We're past feature freeze. Does it need to ship in 4.12? If so, please reference a jira in the PR's title (although, it looks like prow is not actually enforcing FF yet?).

**Comment by mdbooth** [comment]
Not strictly, no.

**Comment by pierreprinetti** [comment]
/hold
Targeted at 4.13.

**Comment by pierreprinetti** [comment]
/hold cancel
ping @mandre @mdbooth 
/meow

**Review by mandre — COMMENTED** [review]
e2e-openstack-nfv-intel failure is unrelated.
/lgtm
/approve

**Pull request merged**